packages/client/src/views/screensharing-modal-puzzel.ts

```diff
@@ -0,0 +1,148 @@
+import { param } from "../params";
+import clsInputs from "../styles/inputs.module.css";
+import { isDialogDefined } from "../helpers/dialog-util";
+import { analyticsEvent } from "../analytics/analytics";
+
+let scriptLoaded: Promise<void> | undefined;
+
+/**
+ * ETTER TESTING:
+ * 1. Fikse den avslutt-chat boksen
+ * 2. Sjekke config-parameterene som sendes inn, customerID, queueKey, interactionId
+ */
+const loadScript = (): Promise<void> => {
+    console.log("Loading Puzzel script");
+    if (scriptLoaded) {
+        return scriptLoaded;
+    }
+    const script = document.createElement("script");
+    script.async = true;
+    script.type = "text/javascript";
+    script.src = "https://app-cdn.puzzel.com/public/js/pzl_loader.js";
+    script.setAttribute("id", "pzlModuleLoader");
+    script.setAttribute("data-customer-id", "41155");
+    const promise = new Promise<void>((resolve) => {
+        script.onload = () => {
+            resolve();
+        };
+    });
+    scriptLoaded = promise;
+    document.body.appendChild(script);
+    return promise;
+};
+
+function lazyLoadScreensharing(openModal: () => void) {
+    console.log("Lazy loading puzzel screensharing");
+    // Check if it is already loaded to avoid layout shift
+    const enabled =
+        window.__DECORATOR_DATA__.params.shareScreen &&
+        window.__DECORATOR_DATA__.features["dekoratoren.skjermdeling"];
+
+    if (!enabled || window.pzl?.info?.status === "started") {
+        openModal();
+        return;
+    }
+    console.log("Screensharing enabled, loading puzzel script");
+    loadScript().then(() => {
+        openModal();
+    });
+}
+
+function startCall(code: string) {
+    window.pzl?.api.showInteraction({
+        interactionId: "7d05c34f-0db4-4fc3-b370-9eb1812a40ca",
+        queueKey: "q_cobrowsing_demo",
+        formValues: {
+            pzlStartChatCode: code,
+        },
+    });
+
+    analyticsEvent({
+        eventName: "skjermdeling",
+        kategori: "dekorator-footer",
+        komponent: "ScreensharingModal",
+    });
+}
+
+export class ScreensharingModalPuzzel extends HTMLElement {
+    dialog!: HTMLDialogElement;
+    input!: HTMLInputElement;
+    errorList!: HTMLElement;
+
+    showModal() {
+        this.dialog.showModal();
+        analyticsEvent({
+            eventName: "modal åpnet",
```

Comment:
fryktelig irriterende at det var fire forskjellige standarder for events her 😂 antageligvis ikke din feil, men en bemerkning

Reply:
Mulig jeg bare er sliten på slutten av dagen, men hva mener du med standarder for events? 😅 
Er det de forskjellige eventNamene som blir sendt med til analyticsEvent du tenker på? For det er jo det faktiske eventet som trigges, og navnene på de følger taksonomien, og resterende variabler er jo like for analyticsEventene.